### PR TITLE
Add VoikkoSpellService Latest

### PR DESCRIPTION
**Pull request by walokra**
VoikkoSpellService is an open source spell-checking service for Finnish on Mac OS X.

### CODE REVIEW

File: Casks/voikkospellservice.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  sha256 '2c01a3e44a0a689763e85b8478e7c13659e5d784336916ba9f7c7f0f36cce159'
+
+  url 'http://verteksi.net/files/osxspell/VoikkoSpellService-1.0.5b1.dmg'
+  name 'VoikkoSpellService'
+  homepage 'http://verteksi.net/lab/osxspell/'
+  license :gpl
+
+  service 'VoikkoSpellService.app'
+
+  zap :delete => [ '~/Library/Services/VoikkoSpellServices.app' ]
```
Comment (jawshooah) [inline Casks/voikkospellservice.rb]: This line is unnecessary. `brew cask uninstall voikkospellservice` will remove the file for you.

File: Casks/voikkospellservice.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+  preflight do
+    system '/usr/bin/killall', '-kill', 'VoikkoSpellService'
+  end
+  uninstall_preflight do
```
Comment (jawshooah) [inline Casks/voikkospellservice.rb]: Please use `uninstall :quit => 'org.puimula.VoikkoSpellService'` instead. Here's an [example](https://github.com/caskroom/homebrew-cask/blob/325b4dd13020b197ef7a858f50581504280350f6/Casks/crashcrier.rb#L12).

File: Casks/voikkospellservice.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  url 'http://verteksi.net/files/osxspell/VoikkoSpellService-1.0.5b1.dmg'
+  name 'VoikkoSpellService'
+  homepage 'http://verteksi.net/lab/osxspell/'
+  license :gpl
+
+  service 'VoikkoSpellService.app'
+
+  zap :delete => [ '~/Library/Services/VoikkoSpellServices.app' ]
+
+  preflight do
```
Comment (jawshooah) [inline Casks/voikkospellservice.rb]: I don't think this `preflight` block is necessary.

File: Casks/voikkospellservice.rb (hunk 2 above)
Comment (walokra) [inline Casks/voikkospellservice.rb]: Didn't get that to work, it just timeouts whereas the killall works.

File: Casks/voikkospellservice.rb (hunk 3 above)
Comment (walokra) [inline Casks/voikkospellservice.rb]: It's there to ensure that the service can be installed/updated.

File: Casks/voikkospellservice.rb (hunk 2 above)
Comment (jawshooah) [inline Casks/voikkospellservice.rb]: Have you tried `uninstall :signal`? ([docs](https://github.com/caskroom/homebrew-cask/blob/master/doc/CASK_LANGUAGE_REFERENCE.md#uninstall-key-signal))

File: Casks/voikkospellservice.rb (hunk 3 above)
Comment (jawshooah) [inline Casks/voikkospellservice.rb]: We don't currently have an upgrade feature (but it's [in the works](https://github.com/caskroom/homebrew-cask/issues/4678)). The best way to upgrade at the moment is to `brew cask uninstall --force` and then `brew cask install`, which as a side effect would kill any existing process using the `uninstall :signal` stanza mentioned [above](https://github.com/caskroom/homebrew-cask/pull/12893#discussion_r35888221).

Comment (jawshooah) [inline Casks/voikkospellservice.rb]: @vitorgalvao, do you have any thoughts on whether this block should remain?

Comment (walokra) [inline Casks/voikkospellservice.rb]: Uninstall and then install is enough I think and this block isn't necessary.

File: Casks/voikkospellservice.rb (hunk 2 above)
Comment (walokra) [inline Casks/voikkospellservice.rb]: Hmm, got it now working with signal.

File: Casks/voikkospellservice.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+  url 'http://verteksi.net/files/osxspell/VoikkoSpellService-1.0.5b1.dmg'
+  name 'VoikkoSpellService'
+  homepage 'http://verteksi.net/lab/osxspell/'
+  license :gpl
+
+  service 'VoikkoSpellService.app'
+
+  uninstall :signal => ['TERM', 'org.puimula.VoikkoSpellService']
+
```
Comment (jawshooah) [inline Casks/voikkospellservice.rb]: Please remove this empty line.

**Comment by jawshooah** [comment]
Thank you. Could you please [squash your commits](http://davidwalsh.name/squash-commits-git)?

**Comment by jawshooah** [comment]
Thank you for the contribution. It was merged directly as 8e1d595a09a06054bc02b788d4b0616528023453 to keep commit history cleaner. Your contribution is still credited to you.

**Pull request closed without merging**